iFixit/react-commerce
# Upgrade to Next.js 12

The project has been updated to Next.js 12 and is now using the [SWC compiler](https://nextjs.org/docs/advanced-features/compiler).
Jest tests however are still run via `babel-jest` / `next/babel`.

This is due to the fact that SWC strictly adhere to the ESModules specification so, according to how a module is declared, we may have difficulties in mocking up functions.   Here is an example:

<img width="1308" alt="Screenshot 2022-08-11 at 00 33 09" src="https://user-images.githubusercontent.com/7805759/184032049-a17f9b89-ec89-4e2a-9a58-03a42d1abd35.png">

 Function 1 is exported via named export, but since it is imported with a namespace notation, it ends up being not modifiable. So any mocking will fail.

This is tracked in several issues: 
https://github.com/swc-project/swc/issues/5205
https://github.com/swc-project/swc/issues/3843#issuecomment-1061382158
https://github.com/facebook/jest/issues/9430

There seems to be a [plugin](https://github.com/swc-project/swc/discussions/5151) specifically designed to solve this issue but currently overriding .swcrc in Next [is not admitted](https://github.com/vercel/next.js/discussions/30413#discussioncomment-1543513)

For those reasons I decided to - at least temporarily - continue using Babel for the Jest transform

cc @ardelato 

## QA
1. Visit the Vercel preview
2. Verify that there are no inconsistencies between the old Next.js 11 version and this version
